### PR TITLE
Import material from model

### LINKED ISSUE #352: When importing a FBX i want to get a mesh and a material resource.

### LINKED ISSUE #466: Add model options gone
When adding a mesh previous options are gone and model will not load

### PR DESCRIPTION
**Pull request by Chainsawkitten**
Fixes model importing. Also adds the option to import materials with the model.

Maya doesn't have the same view as our engine, thus the following mapping applies:
Color channel > Albedo
Bump Mapping channel > Normal map
Specular color channel > Roughness
Reflected color channel > Metallic

Fixes #352 
Fixes #466

### CODE REVIEW

File: src/Editor/Util/AssetConverter.cpp [hunk 1]
```diff
@@ -30,6 +29,25 @@ void AssetConverter::Convert(const char * filepath, const char * destination, gl
         Log() << aImporter.GetErrorString() << "\n";
     }
 
+    if (importMaterial) {
+        /// @todo Fix Windows-specific code.
+        std::string tempMaterialFilePath = filepath;
+        std::string materialFilePath = tempMaterialFilePath.substr(0, tempMaterialFilePath.find_last_of('\\'));
+        materialFilePath += "\\";
```
Comment (NinjaDanz3r) [inline src/Editor/Util/AssetConverter.cpp]: MINOR Variable 'materialFilePath' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

File: src/Editor/Util/AssetConverter.cpp [hunk 2]
```diff
@@ -30,6 +29,25 @@ void AssetConverter::Convert(const char * filepath, const char * destination, gl
...
+
+        std::string tempMaterialDestinationFilePath = destination;
+        std::string materialDestinationFilePath = tempMaterialDestinationFilePath.substr(0, tempMaterialDestinationFilePath.find_last_of('.'));
+        materialDestinationFilePath += "_";
```
Comment (NinjaDanz3r) [inline src/Editor/Util/AssetConverter.cpp]: MINOR Variable 'materialDestinationFilePath' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ModelEditor.cpp#L17](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/4b3491179b2b5df66e11a1b7eb992d20ff916583/src/Editor/GUI/Editors/ModelEditor.cpp#L17): Member variable 'ModelEditor::folder' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/Util/AssetConverter.cpp [hunk 3]
```diff
@@ -30,6 +29,17 @@ void AssetConverter::Convert(const char * filepath, const char * destination, gl
         Log() << aImporter.GetErrorString() << "\n";
     }
 
+    if (importMaterial) {
+        if (aScene->mMeshes[0]->mMaterialIndex >= 0) {
+            aiMaterial* material = aScene->mMaterials[aScene->mMeshes[0]->mMaterialIndex];
+            
+            LoadMaterial(material, aiTextureType_DIFFUSE, materials.albedo);
```
Comment (NinjaDanz3r) [inline src/Editor/Util/AssetConverter.cpp]: MINOR Variable 'materialFilePath' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

File: src/Editor/Util/AssetConverter.cpp [hunk 4]
```diff
@@ -30,6 +29,17 @@ void AssetConverter::Convert(const char * filepath, const char * destination, gl
...
+            LoadMaterial(material, aiTextureType_NORMALS, materials.normal);
+            LoadMaterial(material, aiTextureType_SPECULAR, materials.roughness);
+            LoadMaterial(material, aiTextureType_REFLECTION, materials.metallic);
+        }
```
Comment (NinjaDanz3r) [inline src/Editor/Util/AssetConverter.cpp]: MINOR Variable 'materialDestinationFilePath' is assigned a value that is never used. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunreadVariable)

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ModelEditor.cpp#L17](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/751d505c4a6b2d4ae777487772dee5ed2acbe89b/src/Editor/GUI/Editors/ModelEditor.cpp#L17): Member variable 'ModelEditor::folder' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)
1. ![MINOR][MINOR] [ModelEditor.cpp#L17](https://github.com/Chainsawkitten/LargeGameProjectEngine/blob/751d505c4a6b2d4ae777487772dee5ed2acbe89b/src/Editor/GUI/Editors/ModelEditor.cpp#L17): Member variable 'ModelEditor::folder' is not initialized in the constructor. [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AuninitMemberVar)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/Util/AssetConverter.cpp [hunk 5]
```diff
@@ -46,8 +56,9 @@ std::string& AssetConverter::GetErrorString() {
 }
 
 void AssetConverter::ConvertMeshes(const aiScene * aScene, Geometry::AssetFileHandler * file, glm::vec3 scale, bool flipUVs) {
-    for (unsigned int i = 0; i < aScene->mNumMeshes; ++i)
+    for (unsigned int i = 0; i < aScene->mNumMeshes; ++i) {
```
Comment (SwedishSock) [inline src/Editor/Util/AssetConverter.cpp]: No brackets if for loop is just one row.

**Pull request merged**